packages/client/src/runtime/core/engines/library/LibraryEngine.ts

```diff
@@ -565,3 +575,7 @@ You may have to run ${green('prisma generate')} for your changes to take effect.
     return this.parseEngineResponse(responseString)
   }
 }
+
+function isUserFacingError(e: unknown): e is RequestError['user_facing_error'] {
+  return typeof e === 'object' && e !== null && e['error_code']
```

Comment:
```suggestion
  return typeof e === 'object' && e !== null && e['error_code'] !== undefined
```

so that the return value of the whole function is `true` or `false` and not `undefined` or some truthy value